src/contracts/core/DelegationManager.sol

```diff
@@ -249,19 +263,27 @@ contract DelegationManager is
             withdrawalRoots = new bytes32[](0);
         } else {
             withdrawalRoots = new bytes32[](strategies.length);
+            uint64[] memory totalMagnitudes = allocationManager.getTotalMagnitudes(operator, strategies);
+
             for (uint256 i = 0; i < strategies.length; i++) {
                 IStrategy[] memory singleStrategy = new IStrategy[](1);
                 uint256[] memory singleShare = new uint256[](1);
+                uint64[] memory singleTotalMagnitude = new uint64[](1);
                 singleStrategy[0] = strategies[i];
                 singleShare[0] = shares[i];
+                singleTotalMagnitude[0] = totalMagnitudes[i];
 
                 withdrawalRoots[i] = _removeSharesAndQueueWithdrawal({
                     staker: staker,
                     operator: operator,
-                    withdrawer: staker,
                     strategies: singleStrategy,
-                    shares: singleShare
+                    sharesToWithdraw: singleShare,
+                    totalMagnitudes: singleTotalMagnitude
                 });
+
+                // all shares and queued withdrawn and no delegated operator
+                // reset staker's depositScalingFactor to default
+                depositScalingFactors[staker][strategies[i]] = WAD;
```

Comment:
I can't parse the first line of this comment. maybe just delete it? is it supposed to be justifying why we can / need to reset the staker's depositScalingFactor?